articles/detecting-botnets-using-dns-based-technique/index.md

```diff
@@ -0,0 +1,65 @@
+
+The following diagram shows the structure of a botnet. 
+![alt text](botnet-structure.png)
+
+[Image Source: EC-Council-BLOG](https://blog.eccouncil.org/wp-content/uploads/2018/12/the-structure-of-a-botnet.png)
+
+### Why DNS-Based Techniques Are Widely Used in the Detection of Botnets
+The DNS-based techniques are widely used because of the following reasons:
+
+ * Low cost- the DNS based technique does not require a lot of resources and the cost of the tools is relatively low
+* No effect on network-This detection technique does not affect the network performance or traffic
+
+### DNS-Based Techniques
+
+The DNS-based technique of botnet detection is based on DNS-based network traffic analysis to determine any anomalies. This technique is based on four approaches; failed DNS requests, monitoring malicious domains, domains with low TTLs, and lastly, monitoring abnormal traffic of DNS.
+
+#### 1) Failed DNS requests (NXDOMAIN)
+
+One way of detecting the presence of botnets in a network is by analyzing the [failed DNS resolution requests](http://cesg.tamu.edu/wp-content/uploads/2012/04/reddy_papers/securecomm11.pdf) statistically. Botnets use domains that are not registered, and therefore, their DNS request will fail. However, some botnets such as Torpig utilize low entropy domains so as to dodge being detected. Therefore, a large pool of domains is needed in order to function, and most of these domains fail in their resolution. 
+
+Below is an example of a DNS server request from a non-existing domain.
+ 
+![alt text](failed-dns-request.png)
+
+[Image Source: Super User](https://i.stack.imgur.com/trpaB.png)
+
+#### 2) Monitoring of malicious domains
+
+This technique involves checking all DNS server requests to ensure that none of the domains being resolved is on a blacklist database such as [DNSBL](http://www.dnsbl.info/). Numerous organizations such as [SpamRats](http://www.spamrats.com/) generate a list of domains that are on the blacklist. This becomes the sure way of detecting botnets in a network.  However, there is a disadvantage of this approach since a previous knowledge of the botnet must present and its domain registered. This means it is hard to detect new botnets. 
+
+#### 3) Domains with low TTLs
+
+Botnets creators use a fast-flux technique to hinder detection by modifying the IP address associated with a domain. When the destination IP is changed, detecting faults becomes hard. However, such domains have very low TTL, and this means the DNS system refreshes the resolution cache of the IP related to the domain repeatedly. This, therefore, makes domains with low TTL suspicious. Notably, this technique can be misleading as it can generate false positives due to the many legitimate systems that are connected to the Internet that use such a technique to change IP associated with a domain. 
```

Comment:
Notably, this technique can be misleading as it can generate false positives due to the many legitimate systems that are connected to the Internet that use such a technique to change IP associated with a domain. 

Please simplify this sentence.